src/py-opentimelineio/opentimelineio/plugins/manifest.py

```diff
@@ -3,21 +3,17 @@
 
 """OTIO Python Plugin Manifest system: locates plugins to OTIO."""
 
+from importlib import resources
 import inspect
 import logging
 import os
+from pathlib import Path
 
-# In some circumstances pkg_resources has bad performance characteristics.
-# Using the envirionment variable: $OTIO_DISABLE_PKG_RESOURCE_PLUGINS disables
-# OpenTimelineIO's import and of use of the pkg_resources module.
-if os.environ.get("OTIO_DISABLE_PKG_RESOURCE_PLUGINS", False):
-    pkg_resources = None
-else:
-    try:
-        # on some python interpreters, pkg_resources is not available
-        import pkg_resources
-    except ImportError:
-        pkg_resources = None
+try:
+    from importlib import metadata
+except ImportError:
+    # For python 3.7
+    import importlib_metadata as metadata
```

Comment:
We could theoretically always rely on `importlib_metadata` until `importlib.metadata` stabilizes and we drop 3.7.

Reply:
I think, but can be convinced otherwise, that I'd like to leave it as is.  I like that it's only using the backport when necessary and will otherwise exercise the path moving forward (and gain any performance benefits therein).  Please unresolve if you feel strongly otherwise!

Reply:
I find all the try/except cluttering. As for performance, `importlib_metadata` gets all the performance improvements before they land in CPython.

At least, if we move forward with the current state of the PR, we should limit the install of `importlib_metadata` or Pytohn 3.7 only, using a marker like `importlib_metadata; python_version < "3.8"` in `setup.py`.

Reply:
That seems like a healthy pattern since it ensures that we have lots of proof that the newer module has been working well for a while before we drop the old one. If we rely on the old one, then the drop has more risk. However, @JeanChristopheMorinPerso you mentioned "until importlib.metadata stabilizes" - does that imply that it isn't yet stable? Is there a specific risk there?

Reply:
Limiting the install to only when needed seems like a good change.

Reply:
@jminor `importlib` is stable, so there is no risk. By stabilizing, I meant when they'll stop to change the interface. From what I understand, the interface is "stable" from 3.8 and onward. But I might be mistaken.

Reply:
Good call on the marker, @JeanChristopheMorinPerso.  Added.